docs/_basic/acknowledging_events.md

```diff
@@ -0,0 +1,35 @@
+---
+title: Acknowledging events
+lang: en
+slug: acknowledge
+order: 7
+---
+
+<div class="section-content">
+
+Actions, commands, and options events must **always** be acknowledged using the `ack()` function. This lets Slack know that the event was received and updates the Slack user interface accordingly.
+
+Depending on the type of event, your acknowledgement may be different. For example, when acknowledging a dialog submission you will call `ack()` with validation errors if the submission contains errors, or with no parameters if the submission is valid.
+
+We recommend calling `ack()` right away before sending a new message or fetching information from your database since you only have 3 seconds to respond.
+
+</div>
+
+```python
+# Listen for dialog submissions with a callback_id of ticket_submit
+@app.action("ticket_submit")
```

Comment:
Ah, Bolt for Python doesn't work with code for dialog submission yet (it works when passing a dict to `app.action` or use `app.dialog_submission`). I will create a ticket and will fix this. By the way, should we use a dialog submission example here? I think we may want to use `block_acions` and so on instead.

Reply:
Oh, interesting! I'm not sure why I ported it over this way, but the BoltJS example also uses an object to pass the `callback_id`. I'm happy to change this example to be the following:

```
@app.action({ "callback_id": "ticket_submit" })
```

In terms of changing the example from `block_actions`, I don't feel strongly either way, but I do think we should maintain parity between documentation and change both in one go.

Can you confirm the above suggested change is appropriate and functional?

Reply:
This code works now. You don’t need to change it. https://github.com/slackapi/bolt-python/pull/83

Reply:
After playing around with this for a bit (and understanding what, by definition, are `block_actions`), I'm not sure what example to put in place here. 

We reference the necessity to use `ack()` with errors included as a response when things aren't valid in the leading paragraph, but we give that example (or will, once I push this code, per request) in our `view_submission` example.

Reply:
I think we can use an example of external select menus OR just use a regular acknowledgement. Dialogs have no advantage over modals, so I don't think we should implicitly recommend them with code examples.

Reply:
Went with the external select menu since we reference it in the paragraph. Because we offer this example also in another section, altered it only slightly to draw a parallel to the language used in the paragraph. 

If either of you see a problem with this, or have edits to suggest, just let me know!

```
# Example of responding to an external_select options request
@app.options("menu_selection")
def show_menu_options(ack):
    options = [
        {
            "text": {"type": "plain_text", "text": "Option 1"},
            "value": "1-1",
        },
        {
            "text": {"type": "plain_text", "text": "Option 2"},
            "value": "1-2",
        },
    ]
    ack(options=options)
```

Reply:
Looks good!